### DISCUSSION TITLE
If constants are the default for the := operator, why the const keyword?

### DISCUSSION THREAD

**Discussion by ThatXliner** [vlang/v#8776] [General]
I've read the docs (which are *super* great by the way) which said that constant variables are the default when declaring variables (and assigning). So why do you need "const" if mutability is opt-in?

**Comment by JalonSolov** [comment]
Immutable variables are the default, not constant variables... there's no such thing as a constant variable (those 2 words mean pretty much the opposite).

`const`s can only be initialized at compile time, and there are optimizations that can be performed because of that.

Immutable variables are initialized at runtime, but cannot be changed afterwards.

Mutable variables can be changed as much as you like.

**Reply by ThatXliner** [reply]
Ohhhhh, so it's kind of like [Nim](https://nim-lang.github.io/Nim/tut1.html#constants). Got it! Thanks!

If I'm not mistaken (I'm still learning C++ 😅), that's kind of like C/C++'s macro

```c
#define PI 3.1415
```

**Reply by JalonSolov** [reply]
Effectively similar on the surface, but very different in the internals of the compiler.  :-)  `const` declarations have actual types, etc. where a macro is basically just a text replacement.
